src/plonesocial/microblog/browser/tiles/newpostbox.py

```diff
@@ -40,6 +40,8 @@ def post_context(self):
         ''' The context of this microblog post
         (the portal, a workspace, and so on...)
         '''
+        if self.thread_id:
+            return self.post_container.get(self.thread_id)
         return get_microblog_context(self.context)
 
```

Comment:
@gyst Hm, the idea behind this is: If we have a thread id, then we're posting in the context of a StatusUpdate. That means, the context for the new post is said StatusUpdate. So in other words, we're posting a reply on an existing StatusUpdate. We're not using it as "normal" context, but just as the context in which the message currently being posted lands.

If we don't have a thread id, then it's a new StatusUpdate on its own.

For the case of the activity stream that we show in dashboard and workspace, this seems to do exactly what we want. But I agree, there might lots more under the hood that I don't understand yet.
=> Discuss during the refactoring session.